src/counters.ts

```diff
@@ -0,0 +1,694 @@
+}
+
+/**
+ * Defines an interface to collect the counters for a given process or for the whole system
+ *
+ * @export
+ * @interface ProcessStatistics
+ * @param cpu - percentage (from 0 to 100*vcore).
+ * @param memory - bytes used by the process.
+ * @param ppid - parent process id. undefined for system stats. -1 for totals record for totals record for all the tracked processes.
+ * @param pid - process id. primary key. 0 for system stats. -1 for totals record for all the tracked processes.
+ * @param ctime - ms user + system time.
+ * @param elapsed - ms since the start, of the process/test/system depending on context.
+ * @param timestamp - ms since epoch.
+ */
+export interface ProcessStatisticsCollection {
+	cpu: number[];
+	memory: number[];
+	ppid: number;
+	pid: number;
+	ctime?: number[];
+	elapsed: number[];
+	timestamp: number[];
+}
+
+export interface ProcessStatistics {
+	cpu: number;
+	memory: number;
+	ppid: number;
+	pid: number;
+	ctime?: number;
+	elapsed: number;
+	timestamp: number;
+}
+
+export const ProcessStatisticsUnits: any = {
+	cpu: '%',
+	memory: 'bytes',
+	ctime: 'ms',
+	elapsed: 'ms',
+	timestamp: 'ms'
+}
+
+/**
+ * Defines an interface to specify the counters options for counters tests.
+ * @param collectionInterval - the number of milliseconds after which the counters are collected. Default value is provided by environment variable: CountersCollectionInterval and if undefined then by {@link DefaultCountersOptions}.
+ * @param includeMovingAverages - flag if we should compute movingAverages. Default value is provided by environment variable: CountersIncludeMovingAverages and if undefined then by {@link DefaultCountersOptions}.
+ * @param dumpToFile - flag if counters should be dumped to file. Default value is provided by environment variable: CountersDumpToFile and if undefined then by {@link DefaultCountersOptions}.
+ * @param dumpToChart - flag if counters should be dumped to charts. Default value is provided by environment variable: CountersDumpToChart and if undefined then by {@link DefaultCountersOptions}.
+ * @param outputDirectory - provides path to outputDirectory where output files are written. Default value is provided by environment variable: CountersOutputDirectory and if undefined then by {@link DefaultCountersOptions}.
+ */
+export interface CountersOptions {
+	collectionInterval?: number;
+	includeParent?: boolean;
+	includeMovingAverages?: boolean;
+	dumpToFile?: boolean;
+	dumpToChart?: boolean;
+	outputDirectory?: string;
+}
+
+/**
+ * The default values for CountersOptions.
+ */
+export const DefaultCountersOptions: CountersOptions = { collectionInterval: 200, includeMovingAverages: true, dumpToFile: true, dumpToChart: true, outputDirectory: `${process.env.TEMP}` };
+
+/**
+ * A class with methods that help to implement the counters collection for a test method.
+ */
+export class Counters {
+	static readonly MaxCollectionInterval = 3600 * 1000; // in ms. MaxValue is 1 hour.

```

Comment:
>MaxCollectionInterval [](start = 17, length = 21)

It's usually helpful to have the units be part of the name in situations like this : e.g. MaxCollectionIntervalMs

Reply:
changed.

> > I'm not sure what to review since I don't have a lot of context on this feature but a couple points:
> > 
> > * It seems like this is bringing in a lot of dev dependencies. Whats the plan on how this it used from a day-to-day dev basis? I'm not sure I would like having all these dependencies in my dev environment for ads.
> > * Why are you checking in compiled files? Generally you never check in binaries since it just doubles your change set when making a change.
> 
> I also hate to checkin compiled files. Last time got stuck up with getting it to work without them checked in. Will follow up with you offline to get these done right.

Cleaned up some DevDependencies.